app/parse.py

```diff
@@ -1,9 +1,39 @@
-from dataclasses import dataclass
+import csv
+import time
+from dataclasses import dataclass, fields, astuple
 from urllib.parse import urljoin
 
+from bs4 import BeautifulSoup
+from selenium import webdriver
+from selenium.common.exceptions import (
+    NoSuchElementException,
+    ElementClickInterceptedException, ElementNotInteractableException
+)
+from selenium.webdriver.common.by import By
+from selenium.webdriver.remote.webdriver import WebDriver
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
 
 BASE_URL = "https://webscraper.io/"
-HOME_URL = urljoin(BASE_URL, "test-sites/e-commerce/more/")
+PRODUCT_PAGES = {
+    "home": urljoin(BASE_URL, "test-sites/e-commerce/more/"),
+    "computers": urljoin(BASE_URL, "test-sites/e-commerce/static/computers/"),
+    "laptops": urljoin(BASE_URL, "test-sites/e-commerce/static/computers/laptops"),
+    "tablets": urljoin(BASE_URL, "test-sites/e-commerce/static/computers/tablets"),
+    "phones": urljoin(BASE_URL, "test-sites/e-commerce/static/phones"),
+    "touch": urljoin(BASE_URL, "test-sites/e-commerce/static/phones/touch")
+}
+
+_driver: WebDriver | None = None
+
+
+def get_driver() -> WebDriver:
+    return _driver
```

Comment:
The `get_driver()` function returns `_driver`, which might be `None` if `set_driver()` has not been called. Consider adding a check to ensure `_driver` is not `None` before returning it, or handle the case where it might be `None`.